.github/workflows/docker.yml

```diff
@@ -0,0 +1,122 @@
+name: Build Docker Container
+
+on:
+  push:
+    branches: ["main"]
+    # Publish semver tags as releases.
+    tags: ["*.*.*"]
+  pull_request:
+    branches: ["main"]
+
+env:
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm/v6
+          - linux/arm/v7
+          - linux/arm64
+          - linux/386
+          - linux/ppc64le
+          - linux/s390x
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Convert image name to lowercase
+        run: |
+          GITHUB_REPOSITORY="${{ github.repository }}"
+          echo "IMAGE_NAME=ghcr.io/${GITHUB_REPOSITORY,,}" >>${GITHUB_ENV}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3.0.0
```

Comment:
shall we add setup-qemu-action as mentioned in the docker docs?
https://docs.docker.com/build/ci/github-actions/multi-platform/

Reply:
We are building for all platforms natively (using matrix strategy), so there is no need to add QEMU. QEMU would only be necessary in case we'd be running this job unter linux/amd64 but would like to cross-compile docker images for other platforms.

Reply:
@TrueSkrillor right, so I was thinking that we would want to build linux/arm64 as well, since this would unlock the scanner to be easily runnable from mac os using docker desktop/lima/colima/etc.

Reply:
linux/arm64 is already a target, and builds with the current action.